_posts/2016-09-21-approaching-tdd-outside-android.md

```diff
@@ -0,0 +1,159 @@
+
+That been said, once we got the user story, the next step would be to extract the conditions that the software must satisfy to be accepted, known as the <a href="http://www.seguetech.com/blog/2013/03/25/characteristics-good-agile-acceptance-criteria">acceptance criteria</a>. Those acceptance criteria will define a series of results that must be validated to consider that the feature is done.
+
+<a name="acceptance-criteria"></a>
+
+The acceptance criteria that we have came up with for the <b>“Show account statement”</b> story are:
+
+<b>Scenario 1: Account with transactions</b>
+
+<b>Given</b> the account has the following transactions:
+
+* A deposit of 1000 on 01/04/2014
+* A withdraw of 100 on 02/04/2014
+* A deposit of 500 on 10/04/2014
+
+<b>When</b> the user shows the account statement
+
+<b>Then</b> the statement should be a list with all the transactions in reverse chronological order
+<b>And</b> the statement lines should contain the transaction amount, date and running balance
+
+
+## Setting up the project
+
+Before getting our hands dirty with the code, we have to configure the tools that we are going to use to write our tests. We have chosen <a href="http://junit.org/junit4/">JUnit</a>, <a href="http://mockito.org/">Mockito</a> and <a href="https://google.github.io/android-testing-support-library/docs/espresso/">Espresso</a> for assertions in Android views. We have to add their dependencies to the project build.gradle as follows:
+
+````
+testCompile 'junit:junit:4.12'
+testCompile 'org.mockito:mockito-core:1.10.17'
+
+androidTestCompile "com.android.support.test:runner:0.4.1"
+androidTestCompile "com.android.support.test:rules:0.4.1"
+androidTestCompile "com.android.support.test.espresso:espresso-core:2.2.2"
+androidTestCompile("com.android.support.test.espresso:espresso-contrib:2.2.2") {
+    exclude module: 'recyclerview-v7'
+    exclude module: 'support-v4'
+}
+androidTestCompile 'org.mockito:mockito-core:1.10.17'
+androidTestCompile 'com.google.dexmaker:dexmaker:1.2'
+androidTestCompile 'com.google.dexmaker:dexmaker-mockito:1.2'
+````
+
+And at the bottom of the file:
+
+````
+resolutionStrategy.force "com.android.support:support-annotations:$supportLibraryVersion"
+````
+
+
+For the sake of clarity of this exercise, we are going to use the default source set src/androidTest for our acceptance tests and src/test for the unit tests. This configuration may not be the ideal in a production environment, due to the fact that you would need to add some unit tests for the Android components in your project. If you are in that case, you could end up with a mix of unit/acceptance tests in the same directory, losing the possibility to run all the tests quickly. Remember that acceptance and integration tests are slower than unit tests.
+
+In a real project we will need to run all the unit tests alone to ensure that they are passing and checking the current step in the inner TDD loop. That is why is a good idea to separate them.
```

Comment:
`That is why it's a good idea to separate them.` Maybe add why were are violating that rule here and why it's ok for the exercise.